docs/root/api/http.rst

```diff
@@ -0,0 +1,154 @@
+a previously created `Envoy instance <_starting_envoy>`_.
+
+**Kotlin**::
+
+  val request = RequestBuilder(RequestMethod.POST, "https", "api.envoyproxy.io", "/foo")
+    .addRetryPolicy(RetryPolicy(...))
+    .addHeader("x-custom-header", "foobar")
+    ...
+    .build()
+
+**Swift**::
+
+  let request = RequestBuilder(method: .post, scheme: "https", authority: "api.envoyproxy.io", path: "/foo")
+    .addRetryPolicy(RetryPolicy(...))
+    .addHeader(name: "x-custom-header", value: "foobar")
+    ...
+    .build()
+
+-------------------
+``ResponseHandler``
+-------------------
+
+In order to receive updates for a given request/stream, a ``ResponseHandler`` must be created.
+This class contains a set of callbacks are called whenever an update occurs on the stream.
+
+**Kotlin**::
+
+  val handler = ResponseHandler(Executor {})
+    .onHeaders { headers, statusCode, _ ->
+      Log.d("MainActivity", "Received status: " + statusCode + " and headers: " + headers)
+      Unit
+    }
+    .onData { buffer, endStream ->
+      if (endStream) {
+        Log.d("MainActivity", "Finished receiving body data")
+      } else {
+        Log.d("MainActivity", "Received body data, more incoming")
+      }
+      Unit
+    }
+    .onError { error ->
+      Log.d("MainActivity", "Error received: " + error.message)
+      Unit
+    }
+    ...
+
+**Swift**::
+
+  let handler = ResponseHandler()
+    .onHeaders { headers, statusCode, _ in
+      print("Received status: \(statusCode) and headers: \(headers)")
+    }
+    .onData { buffer, endStream in
+      if endStream {
+        print("Finished receiving body data")
+      } else {
+        print("Received body data, more incoming")
+      }
+    }
+    .onError { error in
+      print("Error received: \(error.message)")
+    }
+    ...
+
+---------------
+``RetryPolicy``
+---------------
+
+The ``RetryPolicy`` type allows for customizing retry rules that should be applied to an outbound
+request. These rules are added by calling ``addRetryPolicy(...)`` on the ``RequestBuilder``, and
+are applied when the request is sent.
+
+For full documentation of how these retry rules perform, see Envoy's documentation:
+
+- `Automatic retries <https://www.envoyproxy.io/learn/automatic-retries>`_
+- `Retry semantics <https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/http_routing.html?highlight=exponential#retry-semantics>`_
+
+-----------------
+``StreamEmitter``
+-----------------
+
+Once a ``Request`` and ``ResponseHandler`` have been created, a stream can be opened using an
+`Envoy instance <_starting_envoy>`_.
+
+Doing so returns a ``StreamEmitter`` which allows the sender to write to the stream,
+close the stream, send trailers, etc.
+
+**Kotlin**::
+
+  val envoy = AndroidEnvoyClientBuilder(...).build()
+
+  val request = RequestBuilder(...).build()
+  val responseHandler = ResponseHandler(...)
+  val emitter = envoy.send(request, responseHandler)
+
+  emitter.sendData(...)
```

Comment:
For these examples, can we follow the builder chained-call pattern as opposed to capturing the `emitter` in a variable?

Reply:
I had originally opted to write it this way to show that data can be sent at a later time by storing the emitter, but I like the chaining callout as well. Updated to exemplify both